packages/core/src/components/Button/Button.jsx

```diff
@@ -58,18 +58,9 @@ export class Button extends React.PureComponent {
 
   classNames() {
     const variationClass = this.props.variation && `ds-c-button--${this.props.variation}`;
-    const disabledClass = this.props.disabled && 'ds-c-button--disabled';
+    const disabledClass = this.props.disabled && this.props.href && 'ds-c-button--disabled';
```

Comment:
We only want to add the `ds-c-button--disabled` class for Buttons with a `href` prop (aka `<a> elements styled like a button), because normally we just use the `disabled` attribute